### ISSUE TITLE
CreateContainerConfigError - secret "airflow-config-envs" not found

### ISSUE THREAD

**Issue by DMilmont** [airflow-helm/charts#486]
## What is the bug?

Running airflow with the KubernetesExecutor - when a pod is launched to run a task it is stuck waiting with a CreateContainerConfigError and message: secret "airflow-config-envs" not found. I am using the default pod template provided in the helm chart. 

I am not sure how to fix this myself unfortunately - would love to hear any advice or work arounds if available. Thank you!

## What version of the chart are you using?

I am using version `8.5.2` of this chart.

## What version of Kubernetes are you using?

```console
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.10", GitCommit:"575467a0eaf3ca1f20eb86215b3bde40a5ae617a", GitTreeState:"clean", BuildDate:"2019-12-11T12:41:00Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.14", GitCommit:"89182bdd065fbcaffefec691908a739d161efc03", GitTreeState:"clean", BuildDate:"2020-12-18T12:02:35Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

## What version of Helm are you using?

```console
version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.17.2"}
```

## What are your custom helm values?

<details>
<summary>click to expand</summary>

```yaml
######## PASTE values.yaml BELOW ########
###################################
# Airflow - Common Configs
###################################
airflow:
  image:
    repository: airflow:2.2.1-python3.8
    pullPolicy: IfNotPresent

  ## the airflow executor type to use
  executor: KubernetesExecutor

  ## the fernet key used to encrypt the connections in the database
  fernetKey: "fernetkey"

  defaultTolerations:
    - key: node-specialization
      operator: Equal
      value: airflow-staging
      effect: NoSchedule


  ## environment variables for the web/scheduler/worker Pods (for airflow configs)
  config:
    # Security
    AIRFLOW__WEBSERVER__EXPOSE_CONFIG: "False"

    # DAGS
    AIRFLOW__CORE__LOAD_EXAMPLES: "True"

    ## Disable noisy "Handling signal: ttou" Gunicorn log messages
    GUNICORN_CMD_ARGS: "--log-level WARNING"

  ## a list of initial users to create
  users:
    - username: admin
      password: admin
      role: Admin
      email: admin@example.com
      firstName: admin
      lastName: admin

  ## a list of initial connections to create
  connections: []
    ## see docs: https://airflow.apache.org/docs/apache-airflow-providers-google/stable/connections/gcp.html
    # - id: my_gcp
    #   type: google_cloud_platform
    #   description: my GCP connection
    #   extra: |-
    #     { "extra__google_cloud_platform__num_retries": "5" }

  ## a list of initial variables to create
  variables:
    - key: "environment"
      value: "dev"

  ## a list of initial pools to create
  pools:
    - name: "pool_1"
      slots: 5
      description: "example pool with 5 slots"
    - name: "pool_2"
      slots: 10
      description: "example pool with 10 slots"

###################################
# Airflow Ingress
###################################
ingress:
  enabled: true
  apiVersion: networking.k8s.io/v1beta1 # Kubernetes 1.18.14
  web:
    host: ""
    annotations:
      kubernetes.io/ingress.class: "nginx-private"
      nginx.ingress.kubernetes.io/enable-cors: "true"
      nginx.ingress.kubernetes.io/cors-allow-methods: "GET, POST"
      nginx.ingress.kubernetes.io/cors-allow-origin: ""
      nginx.ingress.kubernetes.io/cors-allow-credentials: "true"

###################################
# Airflow - Scheduler Configs
###################################
scheduler:
  ## the number of scheduler Pods to run
  replicas: 1

  resources:
    requests:
      cpu: "2000m"
      memory: "1028Mi"

###################################
# Airflow - WebUI Configs
###################################
web:

  ## configs to generate webserver_config.py
  webserverConfig:
    ## the full text value to mount as the webserver_config.py file
    stringOverride: |-
      from flask_appbuilder.security.manager import AUTH_DB
      AUTH_TYPE = AUTH_DB
      AUTH_ROLE_PUBLIC = 'Admin'
      APP_THEME = "bootstrap-theme.css"
  ## the number of web Pods to run
  replicas: 1

  ## resource requests/limits for the airflow web Pods
  resources:
    requests:
      cpu: "200m"
      memory: "2056Mi"

  ## configs for the Service of the web Pods
  service:
    type: ClusterIP
    externalPort: 8080

###################################
# Airflow - Worker Configs
###################################
workers:
  ## if the airflow workers StatefulSet should be deployed
  enabled: false
  replicas: 2

  resources:
    requests:
      cpu: "256m"
      memory: "2Gi"

  podDisruptionBudget:
    enabled: true
    maxUnavailable: "20%"

  autoscaling:
    enabled: true
    maxReplicas: 8
    metrics:
    - type: Resource
      resource:
        name: memory
        target:
          type: Utilization
          averageUtilization: 80

###################################
# Airflow - Flower Configs
###################################
flower:
  ## if the Flower UI should be deployed
  ##
  enabled: false

###################################
# Airflow - Logs Configs
###################################
logs:
  ## configs for the logs PVC
  persistence:
    ## if a persistent volume is mounted at `logs.path`
    enabled: false

###################################
# Kubernetes - Service Account
###################################
serviceAccount:
  create: true
  name: "airflow-staging"

###################################
# Airflow - DAGs Configs
###################################
dags:
  ## configs for the git-sync sidecar (https://github.com/kubernetes/git-sync)
  gitSync:
    ## if the git-sync sidecar container is enabled
    enabled: false

###################################
# Database - PostgreSQL Chart
###################################
postgresql:
  enabled: false

externalDatabase:
  type: postgres
  host: rds_host
  port: 5432
  database: airflow
  user: airflow
  passwordSecret: airflow-postgresql
  passwordSecretKey: postgresql-password

  # use this for any extra connection-string settings, e.g. ?sslmode=disable
  properties: ""

###################################
# Database - Redis Chart
###################################
redis:
  enabled: false

  ...
######## PASTE values.yaml ABOVE ########
```

</details>

**Comment by piyush-daga** [comment]
Might not be too useful, but I too run Airflow on the `KubernetesExecutor` and I checked all my namespaces and did not find the secret - `airflow-config-envs`.

My setup works, the major difference being the airflow version is the one that comes with the chart. Since you're overriding that I'm assuming you need the new features, but you can try with the default version to see if that changes something.

**Comment by thesuperzapper** [comment]
@piyush-daga what namespace are you installing the helm chart into, and is the `Secret/airflow-config-envs` in that namespace?

Note, if you don't specify one, `helm install ...` will use the `default` namespace.

**Comment by thesuperzapper** [comment]
@DMilmont @piyush-daga are you still having this issue?

If so, I imagine that you are somehow submitting your KubernetesExecutor task pods into a DIFFERENT namespace than you have installed airflow into, this won't work because our [`pod_template.yaml` references the `airflow-config-envs` Secret from the airflow install namespace](https://github.com/airflow-helm/charts/blob/f9933599182b835e0211035e946839e094f00195/charts/airflow/files/pod_template.kubernetes-helm-yaml#L54).

**Comment by ldacey** [comment]
I had to add this env var under the airflow.config section:

`AIRFLOW__KUBERNETES__NAMESPACE: "airflow"`

The airflow-config-envs secret was referring to the `default` namespace for some reason, even though my chart is deployed on the `airflow` namespace (using ArgoCD). I happened to check the FERNET_KEY secret in `airflow-config-envs` and it was different than the fernet key secret I created initially. Everything seems to work right now though so maybe that doesn't matter, just seemed odd.
